src/compat/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -0,0 +1,373 @@
+export const description = `
+Tests limitations of bind group usage in a pipeline in compat mode.
+`;
+
+import { makeTestGroup } from '../../../../../common/internal/test_group.js';
+import { keysOf } from '../../../../../common/util/data_tables.js';
+import { ValidationTest } from '../../../../../webgpu/api/validation/validation_test.js';
+import { kRenderEncodeTypes } from '../../../../../webgpu/util/command_buffer_maker.js';
+
+const kTextureTypes = ['regular', 'storage'];
+type TextureType = typeof kTextureTypes[number];
+
+function getTextureTypeWGSL(textureType: TextureType) {
+  return textureType === 'storage' ? 'texture_storage_2d<rgba8unorm, write>' : 'texture_2d<f32>';
+}
+
+type kBindConfigs = ['one bindgroup', 'two bindgroups'];
+type BindConfig = kBindConfigs[number];
+
+/**
+ * Gets the WGSL needed for testing a render pipeline using texture_2d or texture_storage_2d
+ * and either 2 bindgroups or 1
+ */
+function getRenderPipelineWGSL(textureType: TextureType, bindConfig: BindConfig) {
```

Comment:
Both here and `getComputePipelineWGSL` could consider taking the device also and directly returning the module instead of the WGSL.

Reply:
done